docs/kubernetes/deploy-to-google-cloud-from-ci.md

```diff
@@ -0,0 +1,114 @@
+
+## Kubernetes Buildpack
+To deploy project to Google Cloud, you need couple of technologies, to not bother users with installing these technologies, we created a [shopsys/kubernetes-buildpack](https://github.com/shopsys/kubernetes-buildpack) image, which can be used for deployment.
+
+
+## Scripts
+### Deploy to Kubernetes
+Script used for building current state of and application into google cloud
+
+[deploy-to-kubernetes.sh](/project-base/.ci/deploy-to-kubernetes.sh)
+
+#### Environment variables
+| Variable name               | Description
+| -------------               |-------------
+| DOCKER_USERNAME             | docker login
+| DOCKER_PASSWORD             | docker password
+| FIRST_DOMAIN_HOSTNAME       | domain hostname for first domain - without http://
+| SECOND_DOMAIN_HOSTNAME      | domain hostname for second domain - without http://
+| FIRST_DOMAIN_SSL_DIRECTORY  | path to SSL certificates for first domain
+| SECOND_DOMAIN_SSL_DIRECTORY | path to SSL certificates for second domain
+| PROJECT_ID                  | project id of your google project
+| GIT_COMMIT                  | commit hash to build your images uniquely
+
+#### Mounts
+
+##### Docker Socket
+Docker socket is used to build and push image of php-fpm
+```
+-v /var/run/docker.sock:/var/run/docker.sock \
+```
+
+##### Terraform state
+Terraform state mounted locally to be able to apply changes or destroy terraform infrastructure.
+
+```
+-v ~/google-cloud/.terraform/tfstate:/tmp/infrastructure/google-cloud/tfstate \
+```
+
+##### SSL Certificates
+SSL certificates for secured protocol, mount these into folder of your choice that you choose setting environment variable.
+
+```
+-v ~/path/to/certificates-1:$FIRST_DOMAIN_SSL_DIRECTORY \
+-v ~/path/to/certificates-2:$SECOND_DOMAIN_SSL_DIRECTORY \
+```
+
+##### Google Account Service
+Mount your service-account.json obtained from google-cloud
+
+```
+-v ~/google-cloud/service-account.json:/tmp/infrastructure/google-cloud/service-account.json \
+```
+#### Usage
+Use Kubernetes buildpack with environment variable set:
+
+```
+docker run \
+    -v $WORKSPACE/project-base:/tmp \
+    -v /var/run/docker.sock:/var/run/docker.sock \
+    -v ~/google-cloud/.terraform/tfstate:/tmp/infrastructure/google-cloud/tfstate \
+    -v ~/google-cloud/service-account.json:/tmp/infrastructure/google-cloud/service-account.json \
+    -v ~/path/to/certificates-1:$FIRST_DOMAIN_SSL_DIRECTORY \
+    -v ~/path/to/certificates-2:$SECOND_DOMAIN_SSL_DIRECTORY \
+    -e DOCKER_USERNAME \
+    -e DOCKER_PASSWORD \
+    -e GIT_COMMIT \
+    -e FIRST_DOMAIN_HOSTNAME \
+    -e SECOND_DOMAIN_HOSTNAME \
+    -e PROJECT_ID \
+    -e FIRST_DOMAIN_SSL_DIRECTORY \
+    -e SECOND_DOMAIN_SSL_DIRECTORY \
+    --rm \
+    shopsys/kubernetes-buildpack:0.2.0 \
+    .ci/deploy-to-google-cloud.sh
+```
+
+### Destroy Google Cloud Infrastructure
+Script used for deleting infrastructure on google. Script does not need any variable just a mount of Terraform `tfstate` file.
```

Comment:
Should I change something? I guess its the question if this format of documentation is ok. For me it makes most sense, I dont think that more descriptive tutorial would be better. What do you think?

Reply:
maybe this is for @vitek-rostislav because if this is cookbook like he can help you with the formulation of sentences from the actual state to the desired state when we partially says after one step where are we actually now and what will we have after the steps are done

Reply:
I believe this is just a description of the existing script. You could maybe add a sentence describing when to use this, like "if you want to deploy a new version of your application, you need to destroy it first." (I am not saying, that is it the right description, I do not know that :laughing:)